File: 2610-closest-prime-numbers-in-range/2610-closest-prime-numbers-in-range.cpp
class Solution {
public:
    vector<int> closestPrimes(int left, int right) {
        std::vector<int> primeNumbers;
 
        for (int candidate = left; candidate <= right; candidate++) {
            if (candidate % 2 == 0 && candidate > 2) {
                continue;
            }
            if (isPrime(candidate)) { 
                if (!primeNumbers.empty() && candidate <= primeNumbers.back() + 2) {
                    return {primeNumbers.back(), candidate};
                }
                primeNumbers.push_back(candidate);
            }
        }
 
        if (primeNumbers.size() < 2) {
            return {-1, -1};
        }
 
        std::vector<int> closestPair = {-1, -1};
        int minDifference = INT_MAX;
        for (int i = 1; i < primeNumbers.size(); i++) {
            int difference = primeNumbers[i] - primeNumbers[i - 1];
            if (difference < minDifference) {
                minDifference = difference;
                closestPair = {primeNumbers[i - 1], primeNumbers[i]};
            }
        }

        return closestPair;
    }
    private:
    bool isPrime(int number) {
        if (number == 1) {
            return false;
        }
        for (int divisor = 2; divisor <= (int)std::sqrt(number); divisor++) {
            if (number % divisor == 0) {
                return false;
            }
        }
        return true;
    }
};

File: 2610-closest-prime-numbers-in-range/2610-closest-prime-numbers-in-range.java
class Solution {
    public int[] closestPrimes(int left, int right) {
        
        int MAX = 1000000;
        boolean[] isPrime = new boolean[MAX + 1];
        Arrays.fill(isPrime, true);
        isPrime[0] = isPrime[1] = false;
        
        for (int i = 2; i * i <= MAX; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= MAX; j += i) {
                    isPrime[j] = false;
                }
            }
        }
        
        List<Integer> primesInRange = new ArrayList<>();
        for (int i = left; i <= right; i++) {
            if (isPrime[i]) {
                primesInRange.add(i);
            }
        }
        
        if (primesInRange.size() < 2) {
            return new int[] {-1, -1};
        }
        
        int minDiff = Integer.MAX_VALUE;
        int[] result = new int[2];
        
        for (int i = 1; i < primesInRange.size(); i++) {
            int diff = primesInRange.get(i) - primesInRange.get(i - 1);
            if (diff < minDiff) {
                minDiff = diff;
                result[0] = primesInRange.get(i - 1);
                result[1] = primesInRange.get(i);
            }
        }
        
        return result;
    }
}